### PR TITLE
Added the Jarvis march in Coconut

### PR DESCRIPTION
**Pull request by Amaras**

### CODE REVIEW

File: contents/jarvis_march/code/coconut/jarvis_march.coco [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+def counter_clockwise(p1, p2, p3) =
+    (p3.y - p1.y) * (p2.x - p1.x) >= (p2.y - p1.y) * (p3.x - p1.x)
+
+
+def jarvis_march(gift):
+    point_on_hull = min(gift) # The leftmost point in the gift
+    hull = [point_on_hull] # It is guaranteed it will be on the hull.
+
+    while True:
+        # Candidate for the
```
Comment (berquist) [inline contents/jarvis_march/code/coconut/jarvis_march.coco:15]: Incomplete comment?

Comment (Amaras) [reply contents/jarvis_march/code/coconut/jarvis_march.coco]: Incomplete comment indeed.

File: contents/jarvis_march/code/coconut/jarvis_march.coco [hunk 2]
```diff
@@ -0,0 +1,40 @@
+data point(x=0, y=0):
+    def __str__(self):
+        return f'({self.x}, {self.y})'
+
+# Is the turn counter-clockwise?
+def counter_clockwise(p1, p2, p3) =
```
Comment (berquist) [inline contents/jarvis_march/code/coconut/jarvis_march.coco:6]: Is it possible to add an annotation here?

File: contents/jarvis_march/code/coconut/jarvis_march.coco [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    (p3.y - p1.y) * (p2.x - p1.x) >= (p2.y - p1.y) * (p3.x - p1.x)
+
+
+def jarvis_march(gift):
```
Comment (berquist) [inline contents/jarvis_march/code/coconut/jarvis_march.coco:10]: And here?

File: contents/jarvis_march/code/coconut/jarvis_march.coco (hunk 2 above)
Comment (Amaras) [reply contents/jarvis_march/code/coconut/jarvis_march.coco:6]: What do you mean by annotation? `p1 is point`?

File: contents/jarvis_march/code/coconut/jarvis_march.coco (hunk 3 above)
Comment (Amaras) [reply contents/jarvis_march/code/coconut/jarvis_march.coco:10]: would that be something like `def jarvis_march(gift: point[]) -> point[]` or rather something else?

File: contents/jarvis_march/code/coconut/jarvis_march.coco (hunk 2 above)
Comment (berquist) [reply contents/jarvis_march/code/coconut/jarvis_march.coco:6]: Yes.

File: contents/jarvis_march/code/coconut/jarvis_march.coco (hunk 3 above)
Comment (berquist) [reply contents/jarvis_march/code/coconut/jarvis_march.coco:10]: Yes, exactly.

File: contents/jarvis_march/code/coconut/jarvis_march.coco (hunk 2 above)
Comment (Amaras) [reply contents/jarvis_march/code/coconut/jarvis_march.coco:6]: Done

File: contents/jarvis_march/code/coconut/jarvis_march.coco (hunk 3 above)
Comment (Amaras) [reply contents/jarvis_march/code/coconut/jarvis_march.coco:10]: Done

**Review by berquist — APPROVED** [review]

**Pull request merged**